l1-contracts/test/portals/RollupNativeAsset.sol

```diff
@@ -0,0 +1,40 @@
+// SPDX-License-Identifier: Apache-2.0
+// Copyright 2023 Aztec Labs.
+pragma solidity >=0.8.18;
+
+import {ERC20} from "@oz/token/ERC20/ERC20.sol";
+import {DataStructures} from "@aztec/core/libraries/DataStructures.sol";
+import {Registry} from "@aztec/core/messagebridge/Registry.sol";
+
+contract RollupNativeAsset is ERC20 {
+  uint256 public constant P =
```

Comment:
Proposal to make P a constant of `DataStructures` or the likes (ive done so here on an awaiting pr)
https://github.com/AztecProtocol/aztec-packages/blob/f481b3df3916d9177f2cf5342fd14e9e21d87d2f/l1-contracts/src/core/libraries/DataStructures.sol#L7

Reply:
Yep, lets make a `Constants.sol` in there as well. Thinking if it also made sense to look at doing a library for errors, if full library is included in ABI generation, should provide more meaningful errors when simulating tx's in viem 🤔